test/parallel/test-domain-error-types.js

```diff
@@ -6,10 +6,17 @@ const assert = require('assert');
 const domain = require('domain');
 
 // This test is similar to test-domain-multiple-errors, but uses a new domain
-// for each errors.
+// for each error.
 // The test flags are not essential, but serve as a way to verify that
 // https://github.com/nodejs/node/issues/28275 is fixed in debug mode.
 
+// Unfortunately, the test crashes frequently in CI in the debug + container +
+// shared libraries build, so skipping in debug mode for now. This skip can be
+// removed when https://github.com/nodejs/node/issues/30498 is resolved.
+if (process.config.variables.is_debug === 1) {
+  common.skip('temporarily skipping test in debug mode, see issue 30498');
+}
```

Comment:
`[$mode==debug]` in the status file?

Reply:
Ooh, if that will work, yeah, much preferable.

Reply:
It works.

Just a note for anyone looking at the CI results -- the sharedlibs debug CI runs the test runner with `--flaky-tests=skip` which means this test is excluded from the list of tests to run (rather than included and then marked as skipped afterwards):
https://github.com/nodejs/node/blob/35c6e0cc2b56a5380e6808ef5603ecc2b167e032/tools/test.py#L1667-L1668

Reply:
@richardlau That seems … not good? Shouldn’t we run flaky tests in debug mode?

Reply:
> 
> 
> @richardlau That seems … not good? Shouldn’t we run flaky tests in debug mode?

¯\\_(ツ)\_/¯ I don't the history there whether there was a reason to `--flaky-tests=skip` rather than `--flaky-tests=dontcare`. I had a quick GitHub search through https://github.com/nodejs/build but couldn't find anything relevant. I'll switch the debug job to use the existing `IGNORE_FLAKY_TESTS` parameter (and `skip` if true) that the other jobs use and kick off a test build.

cc @nodejs/build FYI

Reply:
Test build: https://ci.nodejs.org/job/node-test-commit-linux-containered/16241/nodes=ubuntu1804_sharedlibs_debug_x64/

Reply:
it was because of this, further up in the jenkins script for debug builds:

```
# see https://github.com/nodejs/node/issues/17016
sed -i 's/\[\$system==linux\]/[$system==linux]\ntest-error-reporting : PASS, FLAKY/g' test/parallel/parallel.status
# see https://github.com/nodejs/node/issues/17017
sed -i 's/\[\$system==linux\]/[$system==linux]\ntest-inspector-async-stack-traces-promise-then : PASS, FLAKY/g' test/sequential/sequential.status
# see https://github.com/nodejs/node/issues/17018
sed -i 's/\[\$system==linux\]/[$system==linux]\ntest-inspector-contexts : PASS, FLAKY/g' test/sequential/sequential.status
```

To get the runner working in the first place we had to manually set those as flaky rather than block on getting them fixed (a choice between not implementing a new builder because it'll be perma-red, finding someone who thinks it's urgent enough to actually fix them, or marking them as special-case flaky and hopefully remove in the future). They could be fixed now but you'll want to also test the other active release lines after commenting those out.

Reply:
It's not that those are marked flaky -- it's that the job was skipping any tests marked flaky rather than running them but not caring about the result (i.e. allowed to fail).